library/ssl_tls13_server.c

```diff
@@ -2954,6 +2964,49 @@ static int ssl_tls13_prepare_new_session_ticket(mbedtls_ssl_context *ssl,
     return 0;
 }
 
+#if defined(MBEDTLS_SSL_EARLY_DATA)
+/* RFC 8446 section 4.2.10
+ *
+ * struct {
+ *     select (Handshake.msg_type) {
+ *         case new_session_ticket:   uint32 max_early_data_size;
+ *         ...
+ *     };
+ * } EarlyDataIndication;
+ */
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_write_nst_early_data_ext(mbedtls_ssl_context *ssl,
```

Comment:
Please adapt `mbedtls_ssl_tls13_write_early_data_ext()` to work in the three cases with an additional parameter probably rather that having a specific one for new session tickets.